### PR TITLE
split data request from cmor csv table

### PR DESCRIPTION
**Pull request by larsbuntemeyer**
for domain dependent data request, we will

- [x] split main data request into table with priority colums (`data-request/dreq_main.csv`): from this table, we create a human readable xlsx, one sheet per priority, more human readable...
- [x] remove priority colum from main cmor csv table (`cmor-table/datasets.csv`): from this table, we will create cmor tables. It's kind of a basic database containing detailed meta data.
- [x] make a check, that each `out_name`, `frequency` combination from any data request in `data-request/*.csv` is available in the `datasets.csv` table.
- [x] add short description to README
- [x] removed all datasets requested by CORDEX Africa from `dreq_main.csv` and created new `dreq_AFR.csv`.

Adding a new data request (for a certain domain) would start by copying the `dreq_main.csv` and modifying or extending it. If it is extended, a new entry in `cmor-table/datasets.csv` would probably be required (would be catched by the test).
Any data request can define their own priorities which are not going into `cmor-table/datasets.csv`.

Example of human readable xlsx derived from `data-request/dreq_main.csv` (one sheet per priority):
[dreq_main.xlsx](https://github.com/user-attachments/files/16235959/dreq_main.xlsx)

Additional domain dependent data requests would go into `data-request/dreq_{domain_id}.csv`.

see also #36

### CODE REVIEW

**Comment by larsbuntemeyer** [comment]
@jesusff @gnikulin 
I implemented here suggestions by @jesusff from #36. Once this is merged, i can add the tables with ocean variables. I think this is a good solution to make the data request independent from cmor tables and make it more humand readable...

File: cmor-table/datasets.csv
Comment (jesusff) [inline cmor-table/datasets.csv:1]: This central file still has the _priority_ column, which would be dependent on the domain. Is it generated automatically? or can we just drop the column manually?

**Review by jesusff — APPROVED** [review]
Thanks! Looks good to me. Just wondering about:
 * the *cell_method* column in the domain-specific tables. Do we expect the same *out_name* and *frequency* requested with different cell_methods in different domains? that would be a nightmare. Or is it there just to remind "humans" of the expected aggregation for each variable? This column complicates the xlsx file with merged cells, but I assume you are trying to mimick the original excel DR, which included the **agg**regation flag (**i**, **a** codes)
 * the naming "main" for the general CORDEX table. We could maybe call it "default", as this would be the basic data request table, unless a particular domain defines their own.

**Comment by jesusff** [comment]
Another couple of related points:
 * We already have a domain-specific (AFR) data request embedded in the main/default data request (search for Africa in the original https://github.com/WCRP-CORDEX/data-request-table/blob/c6603fb/CORDEX-CMIP6/data-request.csv). This simple case could be used for a test.
 * We might have an issue with the comments. These are dropped from the domain-specific CSV but they can be useful to explain the rationale for including a given variable or switching a priority. Currently, there are different uses for the comments. Namely:
    1. Clarify the computation of the variable, as in CMIP. This is discussed in #20 
    2. Clarify the rationale for a variable being requested (e.g. "requested by CORDEX-Africa", "requested for urban modeling")
    3. Warn of CORDEX-specific variables ("not in CMIP or in CF")
    4. Track changes in the variable definitions ("long_name fixed, 2022.09.22")

File: cmor-table/datasets.csv
Comment (larsbuntemeyer) [reply cmor-table/datasets.csv:1]: Thanks, yes, forgot to commit 🤪, that should be removed and only defined in the data request!

**Comment by larsbuntemeyer** [comment]
>     * the _cell_method_ column in the domain-specific tables. Do we expect the same _out_name_ and _frequency_ requested with different cell_methods in different domains? that would be a nightmare. Or is it there just to remind "humans" of the expected aggregation for each variable? This column complicates the xlsx file with merged cells, but I assume you are trying to mimick the original excel DR, which included the **agg**regation flag (**i**, **a** codes)

The latter one, exactly. It's just useful to have the aggregation method in the table to make users think of it depending on the frequency (which they couldn't distinguise when creating the goodle spreadsheets). We could still translate it when creating the xlsx file into something like `i` or `a`. Nevertheless, in the original CORDEX data request, it was the most painful work to figure the cell methods out if it's not provided by the data request in case of new variable frequency combinations. The tests should capture cases when users want different cell methods than in the data request, also, if they use wrong standard names, units, or long names.

>     * the naming "main" for the general CORDEX table. We could maybe call it "default", as this would be the basic data request table, unless a particular domain defines their own.

Agreed, "default" seems to be a better description!

**Comment by larsbuntemeyer** [comment]
> Another couple of related points:
> 
>     * We already have a domain-specific (AFR) data request embedded in the main/default data request (search for Africa in the original https://github.com/WCRP-CORDEX/data-request-table/blob/c6603fb/CORDEX-CMIP6/data-request.csv). This simple case could be used for a test.
> 
>     * We might have an issue with the comments. These are dropped from the domain-specific CSV but they can be useful to explain the rationale for including a given variable or switching a priority.

Ok, yes, these should be restored in the data request, but i think they don't neccessarily need to end up in the cmor table. In the cmor tables, i would prefer the comments to *clarify the computation of the variable*.

**Comment by larsbuntemeyer** [comment]
> We already have a domain-specific (AFR) data request embedded in the main/default data request (search for Africa in the original https://github.com/WCRP-CORDEX/data-request-table/blob/c6603fb/CORDEX-CMIP6/data-request.csv). This simple case could be used for a test.

Done!

**Comment by larsbuntemeyer** [comment]
Ok, let's wait until @gnikulin can have a look at this!

**Comment by larsbuntemeyer** [comment]
@gnikulin i would like to go on with this PR so i can closes some more issue that depend on the new tables structure. Do you think this approach makes sense?

**Pull request merged**